From 5d6fc23716a7de772d79e5d0b098299f525c1cef Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 9 Nov 2015 02:22:16 -0500
Subject: [PATCH 0714/1936] Add show voice list button to ADvoice window. 
 Right window is one click. Make synth engins' c/p/close triplets consistent.

---
 src/UI/ADnoteUI.fl  | 27 ++++++++++++++++++---------
 src/UI/PADnoteUI.fl | 12 ++++++------
 src/UI/SUBnoteUI.fl | 10 +++++-----
 3 files changed, 29 insertions(+), 20 deletions(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index 7b414135..f9b8dfcf 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -1181,9 +1181,9 @@ resui->resonancewindow->show();}
         class ADvoiceUI
       } {}
       Fl_Button {} {
-        label {Close Window}
+        label {Close}
         callback {ADnoteVoice->hide();}
-        xywh {468 601 195 25} box THIN_UP_BOX labelfont 1
+        xywh {700 601 60 25} box THIN_UP_BOX
       }
       Fl_Counter currentvoicecounter {
         label {Current Voice}
@@ -1195,19 +1195,28 @@ advoice->change_voice(nvoice);}
       Fl_Button {} {
         label C
         callback {presetsui->copy(ADnoteVoice->loc(),nvoice);}
-        xywh {705 609 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {630 609 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
         label P
         callback {presetsui->paste(ADnoteVoice->loc(),this,nvoice);}
-        xywh {735 609 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {660 609 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
-        label {Edit ADsynth}
+        label {Show ADsynth}
         callback {ADnoteGlobalParameters->show();}
-        xywh {280 601 105 25} labelfont 1 labelsize 12
+        xywh {280 601 110 25} labelfont 0 labelsize 12
+      }
+      Fl_Button {} {
+        label {Show Voice List}
+        callback {
+	  for (int i=0;i<NUM_VOICES;i++){
+               voicelistitem[i]->refreshlist();
+          }
+          ADnoteVoiceList->show();
+	}
+        xywh {400 601 130 25} labelfont 0 labelsize 12
       }
-
     }
     Fl_Window ADnoteVoiceList {
       label {ADsynth Voices list} open
@@ -1243,9 +1252,9 @@ advoice->change_voice(nvoice);}
         xywh {285 15 25 10} box NO_BOX labelfont 1 labelsize 11
       }
       Fl_Button {} {
-        label {Hide Voice List}
+        label {Close}
         callback {ADnoteVoiceList->hide();}
-        xywh {271 237 125 20}
+        xywh {271 237 80 20}
       }
       Fl_Scroll {} {open
         xywh {0 15 670 220} type VERTICAL box THIN_UP_BOX
diff --git a/src/UI/PADnoteUI.fl b/src/UI/PADnoteUI.fl
index 8fbda582..5069b766 100644
--- a/src/UI/PADnoteUI.fl
+++ b/src/UI/PADnoteUI.fl
@@ -109,7 +109,7 @@ initialized = true;} {}
   } {
     Fl_Window padnotewindow {
       label {PAD synth Parameters} open
-      xywh {52 324 535 435} type Double hide
+      xywh {52 324 535 430} type Double hide
       class Fl_Osc_Window
     } {
 
@@ -835,23 +835,23 @@ if (resui!=NULL) {
 	resui->applybutton->color(FL_GRAY);
 	resui->applybutton->redraw();
 };}
-        xywh {300 400 135 30} box THIN_UP_BOX
+        xywh {200 400 135 25} box THIN_UP_BOX
         code0 {o->color(FL_RED);}
       }
       Fl_Button {} {
         label Close
         callback {padnotewindow->hide();}
-        xywh {440 400 90 30} box THIN_UP_BOX
+        xywh {470 400 60 25} box THIN_UP_BOX
       }
       Fl_Button {} {
         label C
         callback {presetsui->copy(padnotewindow->loc());}
-        xywh {65 400 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {400 405 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
         label P
         callback {presetsui->paste(padnotewindow->loc(),this);}
-        xywh {95 400 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {430 405 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
         label export
@@ -860,7 +860,7 @@ filename=fl_file_chooser("Export samples:","(*.wav)",NULL,0);
 if (filename==NULL) return;
 fl_filename_setext(filename,"");
 o->oscWrite("export2wav", "s", filename);}
-        tooltip {export samples as wav file} xywh {5 400 55 30} box THIN_UP_BOX color 51 labelsize 11 align 128
+        tooltip {export samples as wav file} xywh {5 400 55 25} box THIN_UP_BOX color 51 labelsize 11 align 128
         class Fl_Osc_Button
       }
     }
diff --git a/src/UI/SUBnoteUI.fl b/src/UI/SUBnoteUI.fl
index 8f54cde3..7aec51b5 100644
--- a/src/UI/SUBnoteUI.fl
+++ b/src/UI/SUBnoteUI.fl
@@ -136,7 +136,7 @@ class SUBnoteUI {open : {public PresetsUI_}
   } {
     Fl_Window SUBparameters {
       label {SUBsynth Parameters} open
-      xywh {542 511 735 470} type Double
+      xywh {542 511 735 475} type Double
       class Fl_Osc_Window visible
     } {
       Fl_Box {} {
@@ -155,7 +155,7 @@ class SUBnoteUI {open : {public PresetsUI_}
       Fl_Button {} {
         label Close
         callback {SUBparameters->hide();}
-        xywh {625 446 105 20} box THIN_UP_BOX labelfont 1 labelsize 11
+        xywh {670 446 60 25} box THIN_UP_BOX
       }
       Fl_Group {} {
         label AMPLITUDE open
@@ -326,7 +326,7 @@ freqsettingsui->redraw();}
 h[0]->mag->oscWrite(h[0]->mag->ext, "c", 127);
 h[0]->bw->oscWrite(h[0]->bw->ext, "c", 64);
 SUBparameters->redraw();}
-        tooltip {Clear the harmonics} xywh {445 446 70 20} box THIN_UP_BOX labelfont 1 labelsize 11
+        tooltip {Clear the harmonics} xywh {445 446 70 25} box THIN_UP_BOX
         class Fl_Osc_Button
       }
       Fl_Group bandwidthsettingsui {
@@ -392,12 +392,12 @@ globalfiltergroup->redraw();}
       Fl_Button {} {
         label C
         callback {presetsui->copy(loc);}
-        xywh {540 451 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {600 451 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
         label P
         callback {presetsui->paste(loc,this);}
-        xywh {570 451 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
+        xywh {630 451 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Group {} {
         label OVERTONES open
-- 
2.47.0

